### DISCUSSION TITLE
query parameter to node file, while using fonoster/stream with asterisk.

### DISCUSSION THREAD

**Discussion by CHETAN-SEHGAL** [fonoster/yaps#648] [Q&A]
to use fonoster/stream npm package for audiosocket functionality with asterisk, the dialplan should be - 
exten = 100,1,Verbose("Call to AudioSocket via Dialplan Application")
 same = n,Answer()
 same = n,AudioSocket(uuid,server.example.com:9092)
 same = n,Hangup()
 
 is there a way to pass query parameter in "AudioSocket(uuid,server.example.com:9092)", as i want to send a parameter to the node file using audiosocket. please do let me know if there is any way to solve the problem.

**Comment by psanders** [comment]
The AudioSocket protocol itself doesn't allow for adding parameters.

You will need to set up a channel variable and retrieve its value using ARI, AGI, etc.

(Moving to conversations as it is not an issue but a question.)
